src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -342,32 +459,74 @@ private JButton getNewButton() {
 			String State = InventoryStatus.draft.toString();
 			String user = UserBrowsingManager.getCurrentUser();
 			int checkResults = 0;
-			if (inventoryRowSearchList == null || inventoryRowSearchList.size() < 1) {
-				MessageDialog.error(null, "angal.inventory.noproduct.msg");
+			if (inventoryRowSearchList == null || inventoryRowSearchList.isEmpty()) {
+				MessageDialog.error(null, "angal.inventory.cannotsaveinventorywithoutproducts.msg");
 				return;
 			}
 			LocalDateTime now = LocalDateTime.now();
 			if (dateInventory.isAfter(now)) {
 				MessageDialog.error(null, "angal.inventory.notdateinfuture.msg");
-				return;
+				return ;
```

Comment:
And all similar occurences

```suggestion
				return;
```